core/src/main/java/org/apache/calcite/rel/rules/FilterProjectTransposeRule.java

```diff
@@ -118,6 +121,15 @@ public void onMatch(RelOptRuleCall call) {
     final RelBuilder relBuilder = call.builder();
     RelNode newFilterRel;
     if (copyFilter) {
+      // unknownAsFalse => true since in the WHERE clause:
+      // 1>null evaluates to unknown and WHERE unknown behaves exactly like WHERE false
+      RexSimplify simplifier =
+          new RexSimplify(relBuilder.getRexBuilder(), RelOptPredicateList.EMPTY, true,
```

Comment:
Fun fact: once upon a time RexSimplify was here (it was removed in 1.17.0).
https://github.com/apache/calcite/commit/04cbdb2cead5985322db737e1b0b8ce2259a7306#diff-6766bea931b24cc0e9b6f46456c379c0L123